cloud-consumption/src/main/java/com/sequenceiq/consumption/flow/consumption/storage/handler/StorageConsumptionCollectionHandler.java

```diff
@@ -3,46 +3,100 @@
 import com.sequenceiq.consumption.domain.Consumption;
 import com.sequenceiq.consumption.dto.Credential;
+import com.sequenceiq.consumption.dto.StorageConsumptionResult;
 import com.sequenceiq.consumption.flow.consumption.storage.event.SendStorageConsumptionEvent;
+import com.sequenceiq.consumption.flow.consumption.storage.event.StorageConsumptionCollectionFailureEvent;
 import com.sequenceiq.consumption.flow.consumption.storage.event.StorageConsumptionCollectionHandlerEvent;
+import com.sequenceiq.consumption.service.CloudWatchService;
 import com.sequenceiq.consumption.service.CredentialService;
+import com.sequenceiq.consumption.service.EnvironmentService;
+import com.sequenceiq.environment.api.v1.environment.model.response.DetailedEnvironmentResponse;
 import com.sequenceiq.flow.reactor.api.handler.HandlerEvent;
 
 @Component
 public class StorageConsumptionCollectionHandler  extends AbstractStorageOperationHandler {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(StorageConsumptionCollectionHandler.class);
 
+    private static final int AMOUNT_TO_SUBTRACT = 2;
+
     @Inject
     private CredentialService credentialService;
 
     @Inject
     private CredentialToCloudCredentialConverter credentialConverter;
 
+    @Inject
+    private CloudWatchService cloudWatchService;
+
+    @Inject
+    private EnvironmentService environmentService;
+
     @Override
     public Selectable executeOperation(HandlerEvent<StorageConsumptionCollectionHandlerEvent> event) throws Exception {
         StorageConsumptionCollectionHandlerEvent data = event.getData();
 
         Consumption consumption = data.getContext().getConsumption();
         String environmentCrn = consumption.getEnvironmentCrn();
-        LOGGER.debug("Getting credential for environment with CRN [{}].", environmentCrn);
-        Credential credential = credentialService.getCredentialByEnvCrn(environmentCrn);
-        credentialConverter.convert(credential);
-
         Long resourceId = data.getResourceId();
         String resourceCrn = data.getResourceCrn();
         LOGGER.debug("Storage consumption collection started. resourceCrn: '{}'", resourceCrn);
-        return new SendStorageConsumptionEvent(SEND_CONSUMPTION_EVENT_EVENT.selector(), resourceId, resourceCrn, null);
+
+        try {
+            DetailedEnvironmentResponse detailedEnvironmentResponse = environmentService.getByCrn(environmentCrn);
+            LOGGER.debug("Getting credential for environment with CRN [{}].", environmentCrn);
+            if (detailedEnvironmentResponse.getCloudPlatform().equals(CloudPlatform.AWS.name())) {
+                Credential credential = credentialService.getCredentialByEnvCrn(environmentCrn);
+                CloudCredential cloudCredential = credentialConverter.convert(credential);
+                Date startTime = Date.from(Instant.now().minus(AMOUNT_TO_SUBTRACT, ChronoUnit.DAYS));
+                Date endTime = Date.from(Instant.now());
```

Comment:
Shouldn't the `endTime` be set to the end of the relative previous day?
I assume that we can have the usage for 2 days if the time points to the end of the current day.
cc: @nadamdb @bergerdenes @akanto

Reply:
The goal here is to get the latest data point from CloudWatch for the given bucket. For that we have to set the end time to the current time, so we can get the latest datapoint (e.g. we are running the query at 9am on Friday and the latest data point was on 8am Friday). Theoretically a 24h long period should be enough to get the datapoint for the previous day (so the latest) but there can be some edge cases (probably due to timing inconsistencies) where we have no datapoint in a 24h period so I proposed to use 48h period and extract the latest datapoint from that if we get 2 (or 3).